protocol/src/bistream.rs

```diff
@@ -53,21 +53,21 @@ impl From<(SendStream, RecvStream)> for BiStream {
 }
 
 impl Sink<Frame> for BiStream {
-    type Error = anyhow::Error;
+    type Error = SeliumError;
 
-    fn poll_ready(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
```

Comment:
Is it bad that we define the error type on the `BiStream` and then completely ignore it? I get that the result is the same but seems like a future foot gun...? 🤷‍♂️

Reply:
Updated the `Sink` impls now 👌